feg/gateway/services/session_proxy/servicers/session_controller.go

```diff
@@ -96,14 +96,15 @@ func (srv *CentralSessionController) CreateSession(
 	metrics.PcrfCcrInitRequests.Inc()
 
 	staticRuleInstalls, dynamicRuleInstalls := gx.ParseRuleInstallAVPs(srv.dbClient, gxCCAInit.RuleInstallAVP)
-	chargingKeys := srv.getChargingKeysFromRuleInstalls(staticRuleInstalls, dynamicRuleInstalls)
 	eventTriggers, revalidationTime := gx.GetEventTriggersRelatedInfo(gxCCAInit.EventTriggers, gxCCAInit.RevalidationTime)
 
 	// These rules should not be tracked by OCS or PCRF, they come directly from the orc8r
 	omnipresentRuleIDs, omnipresentBaseNames := srv.dbClient.GetOmnipresentRules()
 	omnipresentRuleIDs = append(omnipresentRuleIDs, srv.dbClient.GetRuleIDsForBaseNames(omnipresentBaseNames)...)
 	staticRuleInstalls = append(staticRuleInstalls, gx.RuleIDsToProtosRuleInstalls(omnipresentRuleIDs)...)
 
+	chargingKeys := srv.getChargingKeysFromRuleInstalls(staticRuleInstalls, dynamicRuleInstalls)
```

Comment:
instead of moving this line, move line 102-104 to line 99 (right after `gx.ParseRuleInstallAVPs`). The reason is this way we will have all the rules collection logic at the same block and all the key, event and timers logic on a different block